src/util/cryptor-http-client.ts

```diff
@@ -0,0 +1,107 @@
+import  { envVars } from "config/env";
+import axios, { AxiosRequestConfig } from "axios";
+import { statsd } from "config/statsd";
+import { cryptorMetrics } from "config/metric-names";
+import Logger from "bunyan";
+
+type CryptorHttpClientOptions = {
+	keyAlias: string;
+	baseUrl?: string;
+	timoutMSec?: number;
+	cryptorChanllenge?: string;
+}
+
+type EncryptionContext = {
+	[key: string]: string | number | undefined | null
+}
+
+type EncryptionPayload = {
+	plainText: string;
+	encryptionContext: EncryptionContext
+}
+
+type DecryptionPayload = {
+	cipherText: string;
+	encryptionContext: EncryptionContext;
+}
+
+/**
+ * This client calling using Cryptor side-car to encrypt/decrypt data.
+ *
+ * How to use:
+ *
+ * - Create Client
+ *   const client = new CryptorHttpClient({keyAlias: "micros/github-for-jira/github-server-app-secrets"});
+ *
+ * - With encryption context (https://developer.atlassian.com/platform/cryptor/concepts/encryption-context/):
+ *   const encrypted = await client.encrypt(req.logger, "super-secret-secret", { "jiraHostname": "https://my-fancy-jira.atlassian.net" });
+ *   const decrypted = await client.decrypt(req.logger, encrypted, { "jiraHostname": "https://my-fancy-jira.atlassian.net" });
+ *
+ * - Without encryption context: Same, but just don't pass the context
+ *   const encrypted = await client.encrypt(req.logger, "super-secret-secret");
+ *
+ */
+export class CryptorHttpClient {
+
+	private readonly keyAlias: string;
+	private readonly axiosCommonConfig: AxiosRequestConfig;
+	private readonly axiosPostConfig: AxiosRequestConfig;
+
+	constructor(opts: CryptorHttpClientOptions) {
+		this.keyAlias = opts.keyAlias;
+		this.axiosCommonConfig = {
+			baseURL: opts.baseUrl || envVars.CRYPTOR_SIDECAR_BASE_URL,
+			timeout: opts.timoutMSec || Number(envVars.CRYPTOR_SIDECAR_TIMEOUT_MSEC)
```

Comment:
Do we really need a configurable timeout here?  Is the timeout even going to ever be triggered?